src/render/canvas/canvas-renderer.ts

```diff
@@ -569,16 +559,12 @@ export class CanvasRenderer extends Renderer {
     }
 
     resizeImage(image: HTMLImageElement, width: number, height: number): HTMLCanvasElement | HTMLImageElement {
-        if (image.width === width && image.height === height) {
```

Comment:
You are going to create a new canvas, even when the width and height are same. Maybe this is only required when image type is a SvgImageElement?